extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -139,19 +139,26 @@ export async function createAzureFunction(connectionString: string, schema: stri
 			});
 
 			// check for the new function file to be created and dispose of the file system watcher
-			const timeoutForFunctionFile = utils.timeoutPromise(constants.timeoutAzureFunctionFileError);
+			const timeoutForFunctionFile = utils.timeoutPromise('');
 			await Promise.race([newFunctionFileObject.filePromise, timeoutForFunctionFile]);
 			propertyBag.quickPickStep = quickPickStep;
 			exitReason = 'finishCreate';
 			TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.finishCreateAzureFunctionWithSqlBinding)
 				.withAdditionalProperties(propertyBag)
 				.withConnectionInfo(connectionInfo).send();
 		} catch (e) {
+			let errorType = utils.getErrorType(e);
 			propertyBag.quickPickStep = quickPickStep;
-			exitReason = 'error';
 			void vscode.window.showErrorMessage(utils.getErrorMessage(e));
 
-			TelemetryReporter.createErrorEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.exitCreateAzureFunctionQuickpick, undefined, utils.getErrorType(e))
+			if (errorType === 'TimeoutError') {
+				// timeout error will only happen if the user cancels the wizard
+				console.log('User cancelled out of the create Azure Function with SQL Binding wizard');
```

Comment:
This is not necessarily true - we don't know why the project may not have been found. A couple other scenarios that we should expect could happen : 

1. The project file is just taking a really long time to be made for some reason (unlikely but possible)
2. An error occurred in the createFunction call but didn't result in an actual error being thrown
3. The project was created but not added to the workspace

So unless we're able to get the AF extension to tell us specifically when the user cancelled out vs anything else happening we should just log a generic message about the search timing out.

Reply:
So there is currently no way to know if a user cancels out of the createFunction API since they created it this way (https://github.com/microsoft/vscode-azurefunctions/issues/2908). 

So the issue seems of not knowing a project file takes a long time and cancels out would end in the same result of the timeoutError happening. 

So the only way to know if a user cancels is to throw during Promise.race() rejects due to the timeout. As such what would you recommend then doing to capture both errors?

Reply:
Both are going to result in a timeout so I don't think there is any way to know. I'm just saying that since we don't know we shouldn't assume it's due to the user cancelling since it could be a number of other reasons why the find timed out.

Reply:
Agreed, so what should I do in this situation. Have a generic error when we get the timeout error - `Unable to Create Azure Function`?

Reply:
Yes, that was my suggestion 😁 

(and by error to be clear I mean log it. We shouldn't be displaying anything to the user right now since I see it being a lot more common that someone will cancel out vs something going wrong and the file not being generated, and so displaying an error would just be annoying and misleading at that point)